File: java/src/main/java/codingdojo/CustomerMatches.java
package codingdojo;

import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.Collection;

public class CustomerMatches {
    private Collection<Customer> duplicates = new ArrayList<>();
    private String matchTerm;
    private Customer customer;

    public Customer getCustomer() {
        return customer;
    }

    public boolean hasDuplicates() {
        return !duplicates.isEmpty();
    }

    public Collection<Customer> getDuplicates() {
        return duplicates;
    }

    public String getMatchTerm() {
        return matchTerm;
    }
}

File: typescript/src/CustomerMatches.ts
import {Customer} from "@/Customer";

export class CustomerMatches {
  readonly duplicates: (Customer | null)[] = []
  matchTerm: string | null = null;
  customer: Customer | null = null;

  hasDuplicates() {
    return this.duplicates.length > 0;
  }

  addDuplicate(duplicate: Customer | null) {
    this.duplicates.push(duplicate);
  }
}
